// Specifies the version of the Protocol Buffers language.
syntax = "proto3";

// Defines a package name to prevent name clashes.
package query;

// The service definition for our query execution service.
service QueryService {
  // A remote procedure call (RPC) for the master to send a sub-query to a worker.
  // It takes a SubQueryRequest message and returns a PartialResult message.
  rpc ExecuteSubQuery(SubQueryRequest) returns (PartialResult);
}

// The message format for the request sent from the master to a worker.
message SubQueryRequest {
  // The raw SQL string to be executed by the worker.
  string query_sql = 1;
}

// The message format for the partial result returned from a worker to the master.
message PartialResult {
  // For now, we'll return the data as a simple JSON string for simplicity.
  string result_json = 1;
}